charmcraft/charm_builder.py

```diff
@@ -239,26 +239,26 @@ def _install_dependencies(self, staging_venv_dir):
         with instrum.Timer("Installing all dependencies"):
             if self.binary_python_packages:
                 # install python packages, allowing binary packages
-                cmd = [pip_cmd, "install", "--upgrade"]  # base command
+                cmd = [pip_cmd, "install"]  # base command
```

Comment:
This change is breaking [centos 7](https://github.com/canonical/charmcraft/actions/runs/5468408751/jobs/9956182984?pr=1136#step:5:239). I'm not sure what the best way to handle it is, so I'm going to tag in @syu-w for thoughts. My best proposal right now is to add a `pip install --upgrade pip setuptools` run before starting any of the conditional installs.